core/src/main/java/com/graphhopper/reader/osm/pbf/PbfBlobDecoder.java

```diff
@@ -326,10 +329,14 @@ private void processOsmPrimitives(byte[] data) throws InvalidProtocolBufferExcep
         PbfFieldDecoder fieldDecoder = new PbfFieldDecoder(block);
```

Comment:
I briefly tried creating the PbfFieldDecoder lazily, only when the current block contains some elements we want to keep, but it made no real difference. So I think unless we do something on a lower level (pbf decoding) this is the earliest we can skip some work here.

Reply:
I wouldn't want to invest a lot of time in this task. The consensus is that the entire input pipeline will need to be rebuilt at some point. At least that's what I've taken away from previous discussions.

Reply:
Yes, this kind of was my point: Skipping this step is easy enough and already yields a nice speed-up.

Reply:
I took another look and apparently the codebase has improved quite a bit since then. Most of my previous pain points have been fixed by https://github.com/graphhopper/graphhopper/pull/2448.

It would still be nice to ditch our forked osmosis parts and use the upstream version as a dependency instead:

https://github.com/graphhopper/graphhopper/blob/53d87aead93ee6753bbb157add63235196bab5e2/core/src/main/java/com/graphhopper/reader/osm/pbf/PbfReader.java#L11-L21

https://github.com/openstreetmap/osmosis/blob/447e5df474584ee2d9ff97b3c26e677968d5331e/osmosis-pbf2/src/main/java/org/openstreetmap/osmosis/pbf2/v0_6/PbfReader.java#L27-L36